src/lib/gadgets/arithmetic.unit-test.ts

```diff
@@ -0,0 +1,68 @@
+import { ZkProgram } from '../proof_system.js';
+import {
+  array,
+  equivalentProvable as equivalent,
+  equivalentAsync,
+  field,
+} from '../testing/equivalent.js';
+import { mod } from '../../bindings/crypto/finite_field.js';
+import { Field } from '../core.js';
+import { Gadgets } from './gadgets.js';
+import { Random } from '../testing/property.js';
+import { provable } from '../circuit_value.js';
+import { assert } from './common.js';
+
+const maybeField = {
+  ...field,
+  rng: Random.map(Random.oneOf(Random.field, Random.field.invalid), (x) =>
+    mod(x, Field.ORDER)
+  ),
+};
```

Comment:
since we do `mod(x, Field.ORDER)` on the result, it gets mapped into the valid range for field elements. so this `maybeField` ends up being the same as `field` with a slightly different random distribution.

I guess, just use `field` instead?